chartpress.py

```diff
@@ -293,14 +305,41 @@ def build_image(image_spec, context_path, dockerfile_path=None, build_args=None)
         "<context_path>/Dockerfile".
     build_args (dict, optional):
         Dictionary of docker build arguments.
+    push (bool, optional):
+        Whether to push the image to a registry
+    builder (str):
+        The container build engine.
+    platforms (list[str], optional):
+        List of platforms to build for
     """
-    cmd = ["docker", "build", "-t", image_spec, context_path]
+    if builder == "docker-build":
+        cmd = ["docker", "build"]
+    elif builder == "docker-buildx":
+        # plain shows the container output, similar to docker build
+        cmd = ["docker", "buildx", "build", "--progress", "plain"]
+    else:
+        raise ValueError(f"Unknown builder: {builder}")
+
+    cmd.extend(["-t", image_spec, context_path])
     if dockerfile_path:
         cmd.extend(["-f", dockerfile_path])
     for k, v in (build_args or {}).items():
         cmd += ["--build-arg", f"{k}={v}"]
+    if platforms:
+        cmd.extend(["--platform", ",".join(platforms)])
+        # Limitations of docker buildx 0.5.1:
+        # - Can't load into the local Docker host and push to a registry at the
+        #   same time
+        # - Can't load multiple platforms into the local docker host
+        if push:
+            cmd.append("--push")
+        elif len(platforms) == 1:
+            cmd.append("--load")
```

Comment:
I love the comments!!! Good you included the version number as well when this was the case.